src/services/AuthService.tsx

```diff
@@ -194,7 +194,13 @@ const Auth = ({ children }: GenericContextProviderProps) => {
 				);
 		};
 
-		if (!initialized || !keycloak.authenticated) return;
+		if (!initialized || !keycloak.authenticated) {
+			if (initialized && !keycloak.authenticated) {
+				logout();
+			}
+
+			return;
+		}
```

Comment:
```suggestion
		if (!initialized) {
		        // keycloak authentication system no initilized (working in demo mode or keycloak not available?)
		        // skipping futher checks (for example: to ask backend if the keycloak token is OK
			return;
		} else {
		        // keycloak authentication is initilized, so it makes sense to check if user is authenticated in keycloak
		        if(!keycloak.authenticated) {
		                // user not authenticated, forcing logout from yaptide app
				logout();		        
		               // skipping futher checks (for example: to ask backend if the keycloak token is OK
		         	return;
		        }
		        // user authenticated, we proceed with further checks
		
		}

		
						if (!initialized || !keycloak.authenticated) {
			if (initialized && !keycloak.authenticated) {
				logout();
			}

			return;
		}
```

Reply:
I checked the double logic and it took me really long while do deal with negations, or and other boolean stuff. The code is working properly, but its very far from being readable.

What do you think of sacrificing the compactness of the code for the readability ?

What about an explicit version with few comments as I suggest above ?

Reply:
Comments are clear and correct. This way is much more readable. I'll add the changes tomorrow.

Reply:
Then I will make some tests again. I was not sure about description of what does `initialized` means, maybe you know better